src/web/components/StickyBottomBanner/BrazeBanner.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState } from 'react';
+import * as emotion from 'emotion';
+import * as emotionCore from '@emotion/core';
+import * as emotionTheming from 'emotion-theming';
+import { onConsentChange } from '@guardian/consent-management-platform';
+import { getZIndex } from '@root/src/web/lib/getZIndex';
+import { Props as BrazeBannerProps } from '@guardian/braze-components';
+import { CanShowResult } from './bannerPicker';
+
+export const brazeVendorId = '5ed8c49c4b8ce4571c7ad801';
+
+type Props = {
+    meta: any;
+};
+
+const containerStyles = emotion.css`
+    position: fixed;
+    bottom: -1px;
+    width: 100%;
+    ${getZIndex('banner')}
+`;
+
+export const hasRequiredConsents = (): Promise<boolean> =>
+    new Promise((resolve) => {
+        onConsentChange(({ tcfv2, ccpa }) => {
+            const consentGivenUnderCcpa = ccpa && !ccpa.doNotSell;
+            const consentGivenUnderTcfv2 =
+                tcfv2 && tcfv2.vendorConsents[brazeVendorId];
+
+            resolve(!!(consentGivenUnderCcpa || consentGivenUnderTcfv2));
+        });
+    });
+
+export const canShow = async (
```

Comment:
This function gets a bit complex and hard to read, partly because of the nested promises going on. It might be worth lifting out some of the promises to make things easier (or defining an inner function to do this). Something like:

```
export const canShow = async (
    hasGivenConsent: boolean,
    isDigitalSubscriber: undefined | boolean,
    brazeUUID: string | null,
): Promise<CanShowResult> => {
    const { brazeSwitch } = window.guardian.config.switches;
    const apiKey = window.guardian.config.page.brazeApiKey;

    if (!(brazeSwitch && apiKey)) {
        return Promise.resolve({ result: false });
    }

    if (isDigitalSubscriber) {
        return Promise.resolve({ result: false });
    }

    // then if hasconsent (promise)
    if (!brazeUUID || !hasGivenConsent) {
        return Promise.resolve({ result: false });
    }

    return import(
        /* webpackChunkName: "braze-web-sdk-core" */ '@braze/web-sdk-core'
    )
        .then(({ default: appboy }) => {
            appboy.initialize(apiKey, {
                noCookies: true,
                baseUrl: 'https://sdk.fra-01.braze.eu/api/v3',
                sessionTimeoutInSeconds: 1,
                minimumIntervalBetweenTriggerActionsInSeconds: 0,
            });

            return appboy.subscribeToInAppMessage((message: any) => {
                const meta = message.extras;

                appboy.changeUser(brazeUUID);
                appboy.openSession();

                if (meta) {
                    return { result: true, meta };
                }

                return { result: false };
            });
        })
        .catch(() => ({ result: false }));
};
```

Even if you don't want to go this far, you can avoid the outer Promise wrapping, by using `Promise.resolve` for the initial early returns.

Reply:
Yeah that's a great point, this could definitely be made more readable by flattening. You're right that the outer Promise wrapping can go. I guess that does make a small difference in that the initial checks go from being async to synchronous but I'm sure that's not a big deal.

Reply:
I ended up re-writing this with `async/await` and i think it's an improvement (ed8da98).